### PR TITLE
demos/advanced-annotations-docs

### PR DESCRIPTION
**Pull request by DJTechnoo**
Created docs about advanced annotations usage.

### CODE REVIEW

**Review by pawelfus — CHANGES_REQUESTED** [review]
We should match button styling to what we already have. For example: http://utils.highcharts.local/samples/#view/stock/navigator/styling-interactive

File: docs/advanced-chart-features/advanced-annotations.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+<iframe style="width: 100%; height: 432px; border: none;" src=https://www.highcharts.com/samples/highcharts/annotations-advanced/elliott-wave allow="fullscreen"></iframe>
+
+Click [here](https://jsfiddle.net/gh/get/library/pure/highcharts/highcharts/tree/master/samples/highcharts/annotations-advanced/elliott-wave) to check the code.
+
+### Crooked line
+<iframe style="width: 100%; height: 432px; border: none;" src=https://www.highcharts.com/samples/highcharts/annotations-advanced/crooked-line allow="fullscreen"></iframe>
+
+Click [here](https://jsfiddle.net/gh/get/library/pure/highcharts/highcharts/tree/master/samples/highcharts/annotations-advanced/crooked-line) to check the code.
+
+### Controllable image
```
Comment (pawelfus) [inline docs/advanced-chart-features/advanced-annotations.md:127]: This and the demo below can be removed - those are not advanced technical annotations, but basic shapes

File: docs/sidebars.js [hunk 2]
```diff
@@ -165,6 +165,7 @@ module.exports = {
         'Advanced chart features': [
             'advanced-chart-features/renderer',
             'advanced-chart-features/annotations-module',
+            'advanced-chart-features/advanced-annotations',
```
Comment (pawelfus) [inline docs/sidebars.js:168]: Move it to Stock category

File: samples/highcharts/annotations-advanced/infinity-line/demo.html [hunk 3]
```diff
@@ -2,3 +2,6 @@
 <script src="https://code.highcharts.com/modules/annotations-advanced.js"></script>
 
 <div id="container"></div>
+<div id="button-box">
+    <button id='applyColors'>Apply colors</button>
```
Comment (pawelfus) [inline samples/highcharts/annotations-advanced/infinity-line/demo.html:6]: It applies colors and some other options. Perhaps change button text?

File: samples/highcharts/annotations-advanced/measure/demo.html [hunk 4]
```diff
@@ -1,4 +1,7 @@
 <script src="https://code.highcharts.com/highcharts.js"></script>
 <script src="https://code.highcharts.com/modules/annotations-advanced.js"></script>
 
-<div id="container1"></div>
+<div id="container"></div>
+<div id="button-box">
+    <button id='applyColors'>Apply colors</button>
```
Comment (pawelfus) [inline samples/highcharts/annotations-advanced/measure/demo.html:6]: Maybe "Change style"? Or "Set theme" here? (+rename button id)

**Review by pawelfus — CHANGES_REQUESTED** [review]
Minor improvements, looks good!

**Review by pawellysy — APPROVED** [review]

**Review by kamil-musialowski — APPROVED** [review]
Great!

File: docs/stock/advanced-annotations.md [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+
+### Tunnel
+<iframe style="width: 100%; height: 432px; border: none;" src=https://www.highcharts.com/samples/highcharts/annotations-advanced/tunnel allow="fullscreen"></iframe>
+
+Click [here](https://jsfiddle.net/gh/get/library/pure/highcharts/highcharts/tree/master/samples/highcharts/annotations-advanced/tunnel) to check the code.
+
+### Time cycles
+<iframe style="width: 100%; height: 432px; border: none;" src=https://www.highcharts.com/samples/highcharts/annotations-advanced/time-cycles allow="fullscreen"></iframe>
+
+Click [here](https://jsfiddle.net/gh/get/library/pure/highcharts/highcharts/tree/master/samples/highcharts/annotations-advanced/time-cycles) to check the code.
```
Comment (kamil-musialowski) [inline docs/stock/advanced-annotations.md:91]: @pawelfus Do we need those? Users can go to JSFiddle via a button which is always displayed in our iframes

Comment (pawelfus) [reply docs/stock/advanced-annotations.md:91]: Good point 👍

Comment (pawelfus) [reply docs/stock/advanced-annotations.md:91]: @pawellysy

File: samples/highcharts/annotations-advanced/tunnel/demo.js [hunk 6]
```diff
@@ -1,22 +1,23 @@
...
-        events: {
-            load: function () {
-                this.annotations.forEach(function (annotation) {
-                    annotation.setControlPointsVisibility(true);
-                    annotation.cpVisibility = true;
-                });
-            }
-        }
-    },
+const chart = Highcharts.chart('container', {
```
Comment (pawelfus) [inline samples/highcharts/annotations-advanced/tunnel/demo.js:1]: How about we add the title like in other updated demos?

File: samples/highcharts/annotations-advanced/fibonacci-reversed/demo.js [hunk 7]
```diff
@@ -1,4 +1,7 @@
 Highcharts.chart('container', {
+    title: {
+        text: 'fibonacci normal and reversed'
```
Comment (pawelfus) [inline samples/highcharts/annotations-advanced/fibonacci-reversed/demo.js:3]: `Fibonacci annotation:`*

**Review by pawelfus — CHANGES_REQUESTED** [review]
I just realized the mistakes in charts titles. The rest looks great!

**Review by pawelfus — APPROVED** [review]
Thanks!

**Review by bm64 — APPROVED** [review]

**Review by TorsteinHonsi — APPROVED** [review]
Thanks, looks good!

**Pull request merged**